pkg/kube/config_getter_test.go

```diff
@@ -78,3 +78,38 @@ func TestImpersonatingConfigGetterNoPrincipalInContext(t *testing.T) {
 		t.Fatalf("incorrect client config:\n%s", diff)
 	}
 }
+
+func TestConfigWithPrincipal(t *testing.T) {
+	user := &auth.UserPrincipal{
+		ID:     "user-id",
+		Groups: []string{"group1", "group2"},
+	}
+	config := &rest.Config{
+		Host: "https://example.com",
+	}
+
+	oidcPrefixes := kube.OIDCPrefixes{
+		UsernamePrefix: "prefix-",
+		GroupsPrefix:   "prefix-",
+	}
+
+	cfg := kube.ConfigWithPrincipal(user, config, oidcPrefixes)
+
+	expectedUser := "prefix-user-id"
+	if cfg.Impersonate.UserName != expectedUser {
+		t.Fatalf("cfg username didn't match expected: %s", cfg.Impersonate.UserName)
+	}
+
+	expectedGroups := []string{"prefix-group1", "prefix-group2"}
+	if diff := cmp.Diff(expectedGroups, cfg.Impersonate.Groups); diff != "" {
+		t.Fatalf("cfg groups didn't match expected:\n%s", diff)
+	}
+
+	// Test with multiple calls.
```

Comment:
You don't need to do this 10 times, you can just do it again :-)

Reply:
Yes thank you for point this out 🙂 🙏